tuf_conformance/test_file_download.py

```diff
@@ -234,7 +259,11 @@ def test_download_with_unknown_hash_algorithm(
     del target.hashes["sha512"]
     repo.targets.targets[target_path] = target
     repo.artifacts[target_path] = Artifact(target_content, target)
+    repo.timestamp.version += 1
+    repo.snapshot.version += 1
+    repo.targets.version += 1
     repo.publish([Targets.type, Snapshot.type, Timestamp.type])
+    repo.update_snapshot()
```

Comment:
I won't reiew tests yet but as an example of what I expect to happen once the version bumping happens by default:     

```
repo.publish(Targets.type)
repo.update_snapshot()
```

that should just work

Reply:
I agree. The purpose of that change was to make it clear why we need to bump by default.